datadog/resource_datadog_user.go

```diff
@@ -30,30 +39,36 @@ func resourceDatadogUser() *schema.Resource {
 				Required: true,
 			},
 			"handle": {
-				Type:     schema.TypeString,
-				Required: true,
+				Type:       schema.TypeString,
+				Optional:   true,
+				Deprecated: "This parameter is deprecated and will be removed from the next Major version",
 			},
 			"is_admin": {
 				Type:       schema.TypeBool,
 				Computed:   true,
 				Optional:   true,
-				Deprecated: "This parameter will be replaced by `access_role` and will be removed from the next Major version",
+				Deprecated: "This parameter is replaced by `roles` and will be removed from the next Major version",
```

Comment:
From my read of the code, these deprecated fields will be ignored in the v2 create code path, but the user will get a deprecation warning during plan/apply. 

WDYT of checking if these are set during the create call and error-ing out if so. It means that we'd lose some backwards compatibility, but would avoid users missing expected permissions.

Reply:
Yeah I think it's worth raising an error for those for security reasons.

Reply:
After looking into it, I'll leave like that. I had to keep the default value to access_roles so it messes up with checking the value.

Reply:
Hmm, yeah the default throws an issue here. 👍 Thanks for looking into it